File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) 
    {
        vector<int> ans(nums.size());
        
        int posIndex = 0; // even index
        int negIndex = 1; // odd index
        
        for(int x : nums)
        {
            if(x > 0)
            {
                ans[posIndex] = x;
                posIndex += 2;
            }
            else
            {
                ans[negIndex] = x;
                negIndex += 2;
            }
        }
        
        return ans;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) 
    {
        int[] arr = new int[nums.length];    

        int oddIndex = 1;
        int evenIndex = 0;

        for(int i = 0; i < nums.length; i++)
        {
            if(nums[i] > 0)
            {
                arr[evenIndex] = nums[i];
                evenIndex = evenIndex + 2;
            }
            else
            {
                arr[oddIndex] = nums[i];
                oddIndex = oddIndex + 2;
            }
        }
        return arr;
    }
}